packages/datadog-instrumentations/src/fs.js

```diff
@@ -179,6 +180,23 @@ function createWrapDirAsyncIterator () {
   }
 }
 
+function isInnerCall (innerResource) {
+  if (this === undefined) return true
+
+  if (!this[ddFsParentSym]) {
+    this[ddFsParentSym] = innerResource.asyncId()
```

Comment:
Can we avoid the use of `this[ddFsParentSym]` with a `WeakMap`?
We should prevent adding properties in external object, always is better if we don't dirty the original object.